Tests/SonarQube.TeamBuild.PreProcessor.Tests/Infrastructure/MockSonarQubeServer.cs

```diff
@@ -79,14 +80,20 @@ IEnumerable<string> ISonarQubeServer.GetInstalledPlugins()
             return this.Data.ServerProperties;
         }
 
-        bool ISonarQubeServer.TryGetQualityProfile(string projectKey, string language, out string qualityProfile)
+        bool ISonarQubeServer.TryGetQualityProfile(string projectKey, string projectBranch, string language, out string qualityProfile)
         {
             this.LogMethodCalled();
 
             Assert.IsFalse(string.IsNullOrEmpty(projectKey), "Project key is required");
             Assert.IsFalse(string.IsNullOrEmpty(language), "Language is required");
 
-            QualityProfile profile = this.Data.QualityProfiles.FirstOrDefault(qp => string.Equals(qp.Language, language) && qp.Projects.Contains(projectKey));
+            string projectId = projectKey;
+            if (!String.IsNullOrWhiteSpace(projectBranch))
+            {
+                projectId = projectKey + ":" + projectBranch;
+            }
+
+            QualityProfile profile = this.Data.QualityProfiles.FirstOrDefault(qp => string.Equals(qp.Language, language) && qp.Projects.Contains(projectId));
```

Comment:
Construct a projectId to correctly query the newly branch-specific quality profile mock.